examples/counter-ts/core/reducers/counter.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  DECREMENT_COUNTER,
+  INCREMENT_COUNTER,
+} from "../constants";
+
+export interface CounterInitialState {
+  value: number;
+}
+
+const INITIAL_STATE = {
+  value: 0,
+};
+
+export default function counter(state: CounterInitialState = INITIAL_STATE, action: { type: string }): any {
+  switch (action.type) {
+    case INCREMENT_COUNTER:
+      return Object.assign({}, {
```

Comment:
No need for `Object.assign` here, this is equivalent to `return { value: state.value + 1 }`